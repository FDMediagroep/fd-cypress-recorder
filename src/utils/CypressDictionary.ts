export enum FdEventType {
    ATTRIBUTE_VALUE_CONTAINS = 'attribute-value-contains',
    ATTRIBUTE_VALUE_EQUALS = 'attribute-value-equals',
    ATTRIBUTE_VALUE_EXISTS = 'attribute-value-exists',
    CLEAR_COOKIES = 'clear-cookies',
    CLICK = 'click',
    CONTAINS_TEXT = 'contains-text',
    HOVER = 'hover',
    LOCATION = 'location',
    LOCATION_CONTAINS = 'location-contains',
    EXISTS = 'exists',
    TYPE = 'type',
    VIEWPORT_SIZE = 'viewport-size',
    VISIT = 'visit',
}

export interface FdEvent {
    type: FdEventType;
}

export interface FdAttributeValueEvent extends FdEvent {
    target: string; // CSS Selector
    name: string;
    value: string;
}

export interface FdAttributeExistsEvent extends FdEvent {
    target: string; // CSS Selector
    name: string;
}

export interface FdTypeEvent extends FdEvent {
    target: string; // CSS Selector
    value: string;
}

export interface FdClickEvent extends FdEvent {
    target: string; // CSS Selector
}

export interface FdHoverEvent extends FdEvent {
    target: string; // CSS Selector
}

export interface FdLocationEvent extends FdEvent {
    href: string;
}

export interface FdLocationContainsEvent extends FdEvent {
    value: string;
}

export interface FdVisitEvent extends FdEvent {
    href: string;
}

export interface FdTextContentEvent extends FdEvent {
    target: string; // CSS Selector
    value: string;
}

export interface FdViewportSizeEvent extends FdEvent {
    width: number;
    height: number;
}

export interface FdExistsEvent extends FdEvent {
    target: string; // CSS Selector
}

export type AllFdEvents = FdEvent
    | FdAttributeValueEvent
    | FdAttributeExistsEvent
    | FdClickEvent
    | FdHoverEvent
    | FdLocationEvent
    | FdLocationContainsEvent
    | FdVisitEvent
    | FdTextContentEvent
    | FdViewportSizeEvent
    | FdExistsEvent
    | FdTypeEvent;

export interface Template {
    name: string;
    description?: string;
    events: AllFdEvents[];
}

export const UNIQUE_SELECTOR_OPTIONS = {
    selectorTypes: ['Tag', 'NthChild']
};

export interface Options {
    basicAuth?: boolean;
}

export function getCode(event: AllFdEvents, options?: Options) {
    switch (event.type) {
        case FdEventType.CLEAR_COOKIES:
            return `cy.clearCookies();`;
        case FdEventType.CLICK:
            return `cy.get('${(event as FdClickEvent).target}').should('exist').click();`;
        case FdEventType.CONTAINS_TEXT:
            return `cy.get('${(event as FdTextContentEvent).target}').contains('${(event as FdTextContentEvent).value.replace(new RegExp("'", 'g'), "\\\'")}');`;
        case FdEventType.EXISTS:
            return `cy.get('${(event as FdExistsEvent).target}').should('exist');`;
        case FdEventType.HOVER:
            return `cy.get('${(event as FdHoverEvent).target}').trigger('mouseover');`;
        case FdEventType.LOCATION:
            return `cy.location('href', {timeout: 10000}).should('eq', '${(event as FdLocationEvent).href}');`;
        case FdEventType.LOCATION_CONTAINS:
            return `cy.location('href', {timeout: 10000}).should('contain', '${(event as FdLocationContainsEvent).value}');`;
        case FdEventType.VISIT:
            if (options && options.basicAuth) {
                return `cy.visit('${(event as FdLocationEvent).href}', {auth: {username: Cypress.env('BASIC_USER') || '', password: Cypress.env('BASIC_PASS') || ''}});`;
            } else {
                return `cy.visit('${(event as FdLocationEvent).href}');`;
            }
        case FdEventType.TYPE:
            return `cy.get('${(event as FdTypeEvent).target}').type('${(event as FdTypeEvent).value}');`;
        case FdEventType.VIEWPORT_SIZE:
            return `cy.viewport(${(event as FdViewportSizeEvent).width}, ${(event as FdViewportSizeEvent).height});`;
        case FdEventType.ATTRIBUTE_VALUE_EQUALS:
            return `cy.get('${(event as FdAttributeValueEvent).target}').then((el: any) => el.attr('${(event as FdAttributeValueEvent).name}')).then((attr: string) => expect(attr).to.eq('${(event as FdAttributeValueEvent).value.replace(new RegExp("'", 'g'), "\\\'")}'))`;
        case FdEventType.ATTRIBUTE_VALUE_CONTAINS:
            return `cy.get('${(event as FdAttributeValueEvent).target}').then((el: any) => el.attr('${(event as FdAttributeValueEvent).name}')).then((attr: string) => expect(attr).to.contain('${(event as FdAttributeValueEvent).value.replace(new RegExp("'", 'g'), "\\\'")}'))`;
        case FdEventType.ATTRIBUTE_VALUE_EXISTS:
            return `cy.get('${(event as FdAttributeExistsEvent).target}').should('have.attr', '${(event as FdAttributeExistsEvent).name}');`;
        default:
            return `// ${JSON.stringify(event)}`;
    }
}
