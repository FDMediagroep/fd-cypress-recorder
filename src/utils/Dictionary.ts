import { AllFdEvents,
    Options,
    FdEventType,
    FdClickEvent,
    FdTextContentEvent,
    FdCountEqualsEvent,
    FdExistsEvent,
    FdHoverEvent,
    FdLocationEvent,
    FdLocationContainsEvent,
    FdTypeEvent,
    FdViewportSizeEvent,
    FdAttributeValueEvent,
    FdAttributeExistsEvent
} from "./FdEvents";

/**
 * Converts event to corresponding code and returns the code.
 * In case there is no translation the `event` will be returned as string.
 *
 * Supported events:
 *
 * FdEventType.ATTRIBUTE_VALUE_CONTAINS: check if given element has given attribute with value containing given value
 *
 * FdEventType.ATTRIBUTE_VALUE_EQUALS: check if given element has given attribute with value equaling given value
 *
 * FdEventType.ATTRIBUTE_VALUE_EXISTS: check if given element has given attribute
 *
 * FdEventType.CLEAR_COOKIES: clear browser session cookies
 *
 * FdEventType.CLICK: click an element
 *
 * FdEventType.CONTAINS_TEXT: check if an element contains given text
 *
 * FdEventType.COUNT_EQUALS: check if an element type exists given number of times within parent
 *
 * FdEventType.COUNT_GREATER_THAN: check if an element type exists more than given number of times within parent
 *
 * FdEventType.COUNT_LESS_THAN: check if an element type exists less than given number of times within parent
 *
 * FdEventType.EXISTS: check if element exists
 *
 * FdEventType.HOVER: set the mouseover state on element
 *
 * FdEventType.LOCATION: check if current location addressbar is the same as given value
 *
 * FdEventType.LOCATION_CONTAINS: check if current location value in addressbar contains given value.
 *                                This is useful to bypass unpredictable values in the URL. E.g. session hashes.
 *
 * FdEventType.VISIT: navigate to the given URL
 *
 * FdEventType.TYPE: type a certain text into the focused element
 *
 * FdEventType.VIEWPORT_SIZE: set the viewport size
 *
 * @param event
 * @param options
 */
export function getCodeFromEvent(event: AllFdEvents, options?: Options): string {
    switch (event.type) {
        case FdEventType.ATTRIBUTE_VALUE_CONTAINS:
            return `cy.get('${(event as FdAttributeValueEvent).target}').then((el) => expect(el.attr('${(event as FdAttributeValueEvent).name}')).to.contain('${(event as FdAttributeValueEvent).value.replace(new RegExp("'", 'g'), "\\\'")}'));`;
        case FdEventType.ATTRIBUTE_VALUE_EQUALS:
            return `cy.get('${(event as FdAttributeValueEvent).target}').then((el) => expect(el.attr('${(event as FdAttributeValueEvent).name}')).to.eq('${(event as FdAttributeValueEvent).value.replace(new RegExp("'", 'g'), "\\\'")}'));`;
        case FdEventType.ATTRIBUTE_VALUE_EXISTS:
            return `cy.get('${(event as FdAttributeExistsEvent).target}').should('have.attr', '${(event as FdAttributeExistsEvent).name}');`;
        case FdEventType.CLEAR_COOKIES:
            return `cy.clearCookies();`;
        case FdEventType.CLICK:
            return `cy.get('${(event as FdClickEvent).target}').click();`;
        case FdEventType.CONTAINS_TEXT:
            return `cy.get('${(event as FdTextContentEvent).target}').contains('${(event as FdTextContentEvent).value.replace(new RegExp("'", 'g'), "\\\'")}');`;
        case FdEventType.COUNT_EQUALS:
            return `cy.get('${(event as FdCountEqualsEvent).target}').should('have.length', ${(event as FdCountEqualsEvent).value});`;
        case FdEventType.COUNT_GREATER_THAN:
            return `cy.get('${(event as FdCountEqualsEvent).target}').should('have.length.gt', ${(event as FdCountEqualsEvent).value});`;
        case FdEventType.COUNT_LESS_THAN:
            return `cy.get('${(event as FdCountEqualsEvent).target}').should('have.length.lt', ${(event as FdCountEqualsEvent).value});`;
        case FdEventType.EXISTS:
            return `cy.get('${(event as FdExistsEvent).target}').should('exist');`;
        case FdEventType.HOVER:
            return `cy.get('${(event as FdHoverEvent).target}').trigger('mouseover');`;
        case FdEventType.LOCATION:
            return `cy.location('href', {timeout: 10000}).should('eq', '${(event as FdLocationEvent).href}');`;
        case FdEventType.LOCATION_CONTAINS:
            return `cy.location('href', {timeout: 10000}).should('contain', '${(event as FdLocationContainsEvent).value}');`;
        case FdEventType.VISIT:
            let opt = '';
            if (options) {
                if (options.basicAuth) {
                    opt = `auth: {username: Cypress.env('BASIC_USER') || '', password: Cypress.env('BASIC_PASS') || ''}`;
                }
                if (options.headers && options.headers.length) {
                    let headers = '';
                    options.headers.forEach((h) => {
                        if (h.property !== '') {
                            const prop = h.property.replace(/\\/g, '\\\\').replace(new RegExp('"', 'g'), '\\"');
                            const value = h.value.replace(/\\/g, '\\\\').replace(new RegExp('"', 'g'), '\\"');
                            if (headers) {
                                headers += `, "${prop}": "${value}"`;
                            } else {
                                headers += `"${prop}": "${value}"`;
                            }
                        }
                    });
                    headers = `headers: {${headers}}`;
                    if (opt) {
                        opt = `${opt}, ${headers}`;
                    } else {
                        opt = headers;
                    }
                }
                if (opt) {
                    opt = `, {${opt}}`;
                }
            }
            return `cy.visit('${(event as FdLocationEvent).href}'${opt});`;
        case FdEventType.TYPE:
            return `cy.get('${(event as FdTypeEvent).target}').type('${(event as FdTypeEvent).value}');`;
        case FdEventType.VIEWPORT_SIZE:
            return `cy.viewport(${(event as FdViewportSizeEvent).width}, ${(event as FdViewportSizeEvent).height});`;
        default:
            return `// ${JSON.stringify(event)}`;
    }
}

export function getCode(suite: string, description: string, events: AllFdEvents[], options?: Options) {
    const code: string[] = [
`/**
  * Code generated with Fd Cypress Recorder.
  * https://github.com/FDMediagroep/fd-cypress-recorder
  */

/// <reference types="Cypress" />
describe('${suite}', () => {
  beforeEach(() => {
    cy.clearCookies();
  });

  it('${description}', () => {
`];
    events.forEach((event) => {
        code.push(`    ${getCodeFromEvent(event, options)}\r\n`);
    });
    code.push('  });\r\n');
    code.push('});\r\n');

    return code;
}
