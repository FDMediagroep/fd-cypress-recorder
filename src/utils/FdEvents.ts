export interface Header {
    property: string;
    value: string;
}

export enum FdEventType {
    ATTRIBUTE_VALUE_CONTAINS = 'attribute-value-contains',
    ATTRIBUTE_VALUE_EQUALS = 'attribute-value-equals',
    ATTRIBUTE_VALUE_EXISTS = 'attribute-value-exists',
    CLEAR_COOKIES = 'clear-cookies',
    CLICK = 'click',
    CONTAINS_TEXT = 'contains-text',
    COUNT_EQUALS = 'count-equals',
    COUNT_GREATER_THAN = 'count-greater-than',
    COUNT_LESS_THAN = 'count-less-than',
    HOVER = 'hover',
    LOCATION = 'location',
    LOCATION_CONTAINS = 'location-contains',
    EXISTS = 'exists',
    /**
     * Type text
     */
    TYPE = 'type',
    VIEWPORT_SIZE = 'viewport-size',
    VISIT = 'visit',
}

export interface FdEvent {
    type: FdEventType;
}

export interface FdAttributeValueEvent extends FdEvent {
    target: string; // CSS Selector
    name: string;
    value: string;
}

export interface FdAttributeExistsEvent extends FdEvent {
    target: string; // CSS Selector
    name: string;
}

export interface FdTypeEvent extends FdEvent {
    target: string; // CSS Selector
    value: string;
}

export interface FdClickEvent extends FdEvent {
    target: string; // CSS Selector
}

export interface FdHoverEvent extends FdEvent {
    target: string; // CSS Selector
}

export interface FdLocationEvent extends FdEvent {
    href: string;
}

export interface FdLocationContainsEvent extends FdEvent {
    value: string;
}

export interface FdVisitEvent extends FdEvent {
    href: string;
}

export interface FdTextContentEvent extends FdEvent {
    target: string; // CSS Selector
    value: string;
}

export interface FdViewportSizeEvent extends FdEvent {
    width: number;
    height: number;
}

export interface FdExistsEvent extends FdEvent {
    target: string; // CSS Selector
}

export interface FdCountEqualsEvent extends FdEvent {
    target: string; // CSS Selector
    value: number;
}

export interface FdCountGreaterThanEvent extends FdEvent {
    target: string; // CSS Selector
    value: number;
}

export interface FdCountLessThanEvent extends FdEvent {
    target: string; // CSS Selector
    value: number;
}

export type AllFdEvents = FdEvent
    | FdAttributeValueEvent
    | FdAttributeExistsEvent
    | FdClickEvent
    | FdCountEqualsEvent
    | FdCountGreaterThanEvent
    | FdCountLessThanEvent
    | FdExistsEvent
    | FdHoverEvent
    | FdLocationEvent
    | FdLocationContainsEvent
    | FdTextContentEvent
    | FdTypeEvent
    | FdViewportSizeEvent
    | FdVisitEvent
;

export interface Template {
    name: string;
    description?: string;
    events: AllFdEvents[];
}

export const UNIQUE_SELECTOR_OPTIONS = {
    selectorTypes: ['Tag', 'NthChild']
};

export interface Options {
    basicAuth?: boolean;
    headers?: Header[];
}
