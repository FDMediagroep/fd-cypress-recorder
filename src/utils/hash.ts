declare let Uint8Array: Uint8ArrayConstructor;
/**
 * Convert given ArrayBuffer to HEX-string.
 * @param buffer
 */
export function hexString(buffer: ArrayBuffer) {
    const byteArray = new Uint8Array(buffer);

    const hexCodes = [...byteArray].map((value) => {
        const hexCode = value.toString(16);
        const paddedHexCode = hexCode.padStart(2, '0');
        return paddedHexCode;
    });

    return hexCodes.join('');
}

/**
 * Digest the given message. ArrayBuffer result is returned in a Promise.
 * @param message
 */
export function digestMessage(message: string) {
    const encoder = new TextEncoder();
    const data = encoder.encode(message);
    return window.crypto.subtle.digest('SHA-256', data);
}

/**
 * Returns digest of given message in the form of a HEX-string.
 * @param message
 */
export async function digest(message: string) {
    const arrayBuffer = await digestMessage(message);
    const hex = hexString(arrayBuffer);
    return hex;
}

/**
 * Returns true when the sum of the ArrayBuffer digest of given message is even and false when odd.
 * @param message
 */
export async function isEven(message: string) {
    const arrayBuffer = await digestMessage(message);
    const byteArray = new Uint8Array(arrayBuffer);
    return (
        byteArray.reduce(
            (previousValue: number, currentValue: number) =>
                previousValue + currentValue
        ) %
            2 ===
        0
    );
}
