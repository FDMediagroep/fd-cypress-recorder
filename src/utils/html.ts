/**
 * Embed given string as JavaScript in the user-DOM space.
 * @param fn
 */
export function embed(fn: (text: string) => void) {
    const script = document.createElement("script");
    script.text = `(${fn.toString()})();`;
    document.documentElement.appendChild(script);
}

/**
 * Convert the given HTML-string to a DOM-node.
 * @param html
 */
export function htmlToElement(html: string): ChildNode {
    const template = document.createElement('template');
    html = html.trim(); // Never return a text node of whitespace as the result
    template.innerHTML = html;
    return template.content.firstChild as ChildNode;
}
