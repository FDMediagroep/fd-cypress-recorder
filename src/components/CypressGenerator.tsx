import React, { PureComponent } from "react";
import styled from "styled-components";
import {TextInput} from '@fdmg/fd-inputs';
import { AllFdEvents, getCode } from "../utils/CypressDictionary";
import { ButtonEditorial } from "@fdmg/fd-buttons";
import FdEvent from "./FdEvent";

export interface Props {
    basicAuth: boolean;
    testSuite: string;
    testDescription: string;
    events: AllFdEvents[];
    onRemoveEvent: (index: number) => void;
    onSuiteChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    onDescriptionChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

/**
 * This component can show the recorded events as code or as a list of individual events.
 */
export default class CypressGenerator extends PureComponent<Props, any> {
    state: any = {
        view: 'code'
    };

    constructor(props: any) {
        super(props);
        this.state.cypressCode = this.generateCodeFromEvents(this.props.events);
    }

    componentDidUpdate(prevProps: any) {
        if (prevProps.testSuite !== this.props.testSuite
            || prevProps.testDescription !== this.props.testDescription
            || prevProps.events.length !== this.props.events.length
            || prevProps.basicAuth !== this.props.basicAuth) {
            this.setState({cypressCode: this.generateCodeFromEvents(this.props.events)});
        }
    }

    handleTestSuiteChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        this.props.onSuiteChange(e);
    }

    handleTestDescriptionChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        this.props.onDescriptionChange(e);
    }

    handleChange = () => {
    }

    toggleView = () => {
        this.setState({view: this.state.view === 'code' ? 'events' : 'code'});
    }

    removeEvent = (e: React.MouseEvent<HTMLButtonElement>) => {
        this.props.onRemoveEvent(parseInt(e.currentTarget.getAttribute('data-index') || '0', 10));
    }

    generateCodeFromEvents = (events: AllFdEvents[]) => {
        let code: any[] = events.reduce((resultArray: any[], event: any) => {
            resultArray.push(`${getCode(event, {basicAuth: this.props.basicAuth})}\r\n`);
            return resultArray;
        }, []);

        const suite = this.props.testSuite ? this.props.testSuite.replace(new RegExp("'", 'g'), "\\\'") : 'Test Suite ...';
        const description = this.props.testDescription ? this.props.testDescription.replace(new RegExp("'", 'g'), "\\\'") : 'should ...';
        // Add tabs for existing lines
        code = code.map((line: string) => {
            return `\t\t${line}`;
        });

        code.unshift(...[
            `/**\r\n`,
            ` * Code generated with Fd Cypress Recorder.\r\n`,
            ` * https://github.com/FDMediagroep/fd-cypress-recorder\r\n`,
            ` */\r\n\r\n`,
            `/// <reference types="Cypress" />\r\n`,
            `describe('${suite}', () => {\r\n`,
            `\tafterEach(() => {\r\n`,
            `\t\tcy.clearCookies();\r\n`,
            `\t});\r\n\r\n`,
            `\tit('${description}', () => {\r\n`
        ]);

        code.push('\t});\r\n');
        code.push('});\r\n');

        return code;
    }

    render() {
        return (
            <StyledCodeContainer>
                <TextInput id="testSuite" className="testSuite" type="text" label="Test suite name" onChange={this.handleTestSuiteChange} value={this.props.testSuite ? this.props.testSuite : ''} errorMessage="Please enter a name for your test suite"/>
                <TextInput id="testDescription" className="testDescription" type="text" label="Description" onChange={this.handleTestDescriptionChange} value={this.props.testDescription ? this.props.testDescription : ''} errorMessage="Please enter a description for your test"/>
                {
                    this.state.view === 'code' ?
                    <textarea onChange={this.handleChange} value={this.state.cypressCode.join('')} placeholder="Start using the website to record some events" readOnly={false} spellCheck={false}/>
                    : (
                        <ul>
                            {this.props.events.map((event: AllFdEvents, idx: number) => (
                                <li key={idx}>
                                    <FdEvent event={event}/><ButtonEditorial className="toggle-view" data-index={idx} onClick={this.removeEvent} title="Delete event">x</ButtonEditorial>
                                </li>
                            ))}
                        </ul>
                    )
                }
                <ButtonEditorial className="toggle-view" onClick={this.toggleView} title="Toggle between code and event view">{this.state.view === 'code' ? 'Show events' : 'Show code'}</ButtonEditorial>
            </StyledCodeContainer>
        );
    }
}

const StyledCodeContainer = styled.div`
    display: flex;
    flex: 1 1 auto;
    flex-direction: column;
    min-width: 450px;

    .testSuite, .testDescription {
        width: 100%;
        margin-bottom: 1rem;
    }

    textarea {
        white-space: nowrap;
    }

    ul, textarea {
        flex: 1 1 auto;
        margin-bottom: .5rem;

        ::-webkit-scrollbar {
            width: .5rem;
            height: .5rem;
        }

        ::-webkit-scrollbar-track {
            -webkit-border-radius: .25rem;
            border-radius: .25rem;
            background:rgba(0,0,0,0.1);
        }

        ::-webkit-scrollbar-thumb {
            -webkit-border-radius: .25rem;
            border-radius: .25rem;
            background:rgba(0,0,0,0.2);
        }

        ::-webkit-scrollbar-thumb:hover {
            background: rgba(0,0,0,0.4);
        }

        ::-webkit-scrollbar-thumb:window-inactive {
            background: rgba(0,0,0,0.05);
        }
    }
    ul {
        padding: 0;
        margin: 0;
        list-style-type: none;
        overflow: auto;
        max-height: 290px;
        li {
            display: flex;
            align-items: center;
            justify-content: space-between;
            border-bottom: 1px solid rgba(0, 0, 0, 0.1);
            padding: .5rem 1rem;
        }
    }
    .toggle-view {
        margin-bottom: .5rem;
    }

`;
