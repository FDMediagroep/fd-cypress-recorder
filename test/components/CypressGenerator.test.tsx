import React from 'react';
import TestRenderer from 'react-test-renderer';
import CypressGenerator from '../../src/components/CypressGenerator';
import { FdEventType } from '../../src/utils/CypressDictionary';

describe('Cypress Generator', () => {
    const onSuiteChange = jest.fn();
    const onDescriptionChange = jest.fn();
    const onRemoveEvent = jest.fn();

    it('should render empty test correctly', () => {
        let contextMenu = TestRenderer.create(<CypressGenerator
            basicAuth={false}
            events={[]}
            onSuiteChange={onSuiteChange}
            onDescriptionChange={onDescriptionChange}
            onRemoveEvent={onRemoveEvent}
            testSuite=""
            testDescription=""/>);
        expect(contextMenu.toJSON()).toMatchSnapshot();
    });

    it('should render test with suite name and description correctly', () => {
        let contextMenu = TestRenderer.create(<CypressGenerator
            basicAuth={false}
            events={[]}
            onSuiteChange={onSuiteChange}
            onDescriptionChange={onDescriptionChange}
            onRemoveEvent={onRemoveEvent}
            testSuite="Test suite name"
            testDescription="Test description"/>);
        expect(contextMenu.toJSON()).toMatchSnapshot();
    });

    it('should render events correctly', () => {
        let contextMenu = TestRenderer.create(<CypressGenerator
            basicAuth={false}
            events={[
                {type: FdEventType.VIEWPORT_SIZE, width: 320, height: 240},
                {type: FdEventType.VISIT, href: 'http://willemliu.nl'}
            ]}
            onSuiteChange={onSuiteChange}
            onDescriptionChange={onDescriptionChange}
            onRemoveEvent={onRemoveEvent}
            testSuite=""
            testDescription=""/>);
        expect(contextMenu.toJSON()).toMatchSnapshot();
    });

});
