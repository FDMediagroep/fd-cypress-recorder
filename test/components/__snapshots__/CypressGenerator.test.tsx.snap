// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cypress Generator should render empty test correctly 1`] = `
<div
  className="sc-ifAKCX iILkrY"
>
  <div
    className="fd-input testSuite"
  >
    <div
      className="input"
    >
      <input
        id="testSuite"
        maxLength={null}
        minLength={null}
        name="testSuite"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Test suite name"
        type="text"
        value=""
      />
      <label
        htmlFor="testSuite"
      >
        Test suite name
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      >
        Please enter a name for your test suite
      </div>
      <div
        className="input-message"
      />
    </div>
  </div>
  <div
    className="fd-input testDescription"
  >
    <div
      className="input"
    >
      <input
        id="testDescription"
        maxLength={null}
        minLength={null}
        name="testDescription"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Description"
        type="text"
        value=""
      />
      <label
        htmlFor="testDescription"
      >
        Description
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      >
        Please enter a description for your test
      </div>
      <div
        className="input-message"
      />
    </div>
  </div>
  <textarea
    onChange={[Function]}
    placeholder="Start using the website to record some events"
    readOnly={false}
    spellCheck={false}
    value="/**
  * Code generated with Fd Cypress Recorder.
  * https://github.com/FDMediagroep/fd-cypress-recorder
  */

/// <reference types=\\"Cypress\\" />
describe('Test Suite ...', () => {
  

  afterEach(() => {
    cy.clearCookies();
  });

  it('should ...', () => {
  });
});
"
  />
  <div
    className="sc-EHOje gXMykk"
  >
    <button
      className="fd-button  selected"
      onClick={[Function]}
      title="Show code"
    >
      Code
    </button>
    <button
      className="fd-button"
      onClick={[Function]}
      title="Show events"
    >
      Events
    </button>
    <button
      className="fd-button"
      onClick={[Function]}
      title="Show headers"
    >
      Headers
    </button>
    <label
      className="basic-auth"
    >
      <input
        checked={false}
        onChange={[MockFunction]}
        type="checkbox"
      />
       
      Basic Auth
    </label>
  </div>
</div>
`;

exports[`Cypress Generator should render events correctly 1`] = `
<div
  className="sc-ifAKCX iILkrY"
>
  <div
    className="fd-input testSuite"
  >
    <div
      className="input"
    >
      <input
        id="testSuite"
        maxLength={null}
        minLength={null}
        name="testSuite"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Test suite name"
        type="text"
        value=""
      />
      <label
        htmlFor="testSuite"
      >
        Test suite name
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      >
        Please enter a name for your test suite
      </div>
      <div
        className="input-message"
      />
    </div>
  </div>
  <div
    className="fd-input testDescription"
  >
    <div
      className="input"
    >
      <input
        id="testDescription"
        maxLength={null}
        minLength={null}
        name="testDescription"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Description"
        type="text"
        value=""
      />
      <label
        htmlFor="testDescription"
      >
        Description
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      >
        Please enter a description for your test
      </div>
      <div
        className="input-message"
      />
    </div>
  </div>
  <textarea
    onChange={[Function]}
    placeholder="Start using the website to record some events"
    readOnly={false}
    spellCheck={false}
    value="/**
  * Code generated with Fd Cypress Recorder.
  * https://github.com/FDMediagroep/fd-cypress-recorder
  */

/// <reference types=\\"Cypress\\" />
describe('Test Suite ...', () => {
  beforeEach(() => {
    cy.visit('http://willemliu.nl');
    cy.clearCookies();
    cy.reload(true);
  });

  afterEach(() => {
    cy.clearCookies();
  });

  it('should ...', () => {
    cy.viewport(320, 240);
    cy.visit('http://willemliu.nl');
  });
});
"
  />
  <div
    className="sc-EHOje gXMykk"
  >
    <button
      className="fd-button  selected"
      onClick={[Function]}
      title="Show code"
    >
      Code
    </button>
    <button
      className="fd-button"
      onClick={[Function]}
      title="Show events"
    >
      Events
    </button>
    <button
      className="fd-button"
      onClick={[Function]}
      title="Show headers"
    >
      Headers
    </button>
    <label
      className="basic-auth"
    >
      <input
        checked={false}
        onChange={[MockFunction]}
        type="checkbox"
      />
       
      Basic Auth
    </label>
  </div>
</div>
`;

exports[`Cypress Generator should render headers correctly 1`] = `
<div
  className="sc-ifAKCX iILkrY"
>
  <div
    className="fd-input testSuite"
  >
    <div
      className="input"
    >
      <input
        id="testSuite"
        maxLength={null}
        minLength={null}
        name="testSuite"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Test suite name"
        type="text"
        value=""
      />
      <label
        htmlFor="testSuite"
      >
        Test suite name
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      >
        Please enter a name for your test suite
      </div>
      <div
        className="input-message"
      />
    </div>
  </div>
  <div
    className="fd-input testDescription"
  >
    <div
      className="input"
    >
      <input
        id="testDescription"
        maxLength={null}
        minLength={null}
        name="testDescription"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Description"
        type="text"
        value=""
      />
      <label
        htmlFor="testDescription"
      >
        Description
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      >
        Please enter a description for your test
      </div>
      <div
        className="input-message"
      />
    </div>
  </div>
  <textarea
    onChange={[Function]}
    placeholder="Start using the website to record some events"
    readOnly={false}
    spellCheck={false}
    value="/**
  * Code generated with Fd Cypress Recorder.
  * https://github.com/FDMediagroep/fd-cypress-recorder
  */

/// <reference types=\\"Cypress\\" />
describe('Test Suite ...', () => {
  beforeEach(() => {
    cy.visit('http://willemliu.nl');
    cy.clearCookies();
    cy.reload(true);
  });

  afterEach(() => {
    cy.clearCookies();
  });

  it('should ...', () => {
    cy.viewport(320, 240);
    cy.visit('http://willemliu.nl', {headers: {\\"Content-type\\": \\"application/json\\", \\"x-platform\\": \\"Android\\"}});
  });
});
"
  />
  <div
    className="sc-EHOje gXMykk"
  >
    <button
      className="fd-button  selected"
      onClick={[Function]}
      title="Show code"
    >
      Code
    </button>
    <button
      className="fd-button"
      onClick={[Function]}
      title="Show events"
    >
      Events
    </button>
    <button
      className="fd-button"
      onClick={[Function]}
      title="Show headers"
    >
      Headers
    </button>
    <label
      className="basic-auth"
    >
      <input
        checked={false}
        onChange={[MockFunction]}
        type="checkbox"
      />
       
      Basic Auth
    </label>
  </div>
</div>
`;

exports[`Cypress Generator should render test with suite name and description correctly 1`] = `
<div
  className="sc-ifAKCX iILkrY"
>
  <div
    className="fd-input testSuite"
  >
    <div
      className="input"
    >
      <input
        id="testSuite"
        maxLength={null}
        minLength={null}
        name="testSuite"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Test suite name"
        type="text"
        value="Test suite name"
      />
      <label
        htmlFor="testSuite"
      >
        Test suite name
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      >
        Please enter a name for your test suite
      </div>
      <div
        className="input-message"
      />
    </div>
  </div>
  <div
    className="fd-input testDescription"
  >
    <div
      className="input"
    >
      <input
        id="testDescription"
        maxLength={null}
        minLength={null}
        name="testDescription"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Description"
        type="text"
        value="Test description"
      />
      <label
        htmlFor="testDescription"
      >
        Description
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      >
        Please enter a description for your test
      </div>
      <div
        className="input-message"
      />
    </div>
  </div>
  <textarea
    onChange={[Function]}
    placeholder="Start using the website to record some events"
    readOnly={false}
    spellCheck={false}
    value="/**
  * Code generated with Fd Cypress Recorder.
  * https://github.com/FDMediagroep/fd-cypress-recorder
  */

/// <reference types=\\"Cypress\\" />
describe('Test suite name', () => {
  

  afterEach(() => {
    cy.clearCookies();
  });

  it('Test description', () => {
  });
});
"
  />
  <div
    className="sc-EHOje gXMykk"
  >
    <button
      className="fd-button  selected"
      onClick={[Function]}
      title="Show code"
    >
      Code
    </button>
    <button
      className="fd-button"
      onClick={[Function]}
      title="Show events"
    >
      Events
    </button>
    <button
      className="fd-button"
      onClick={[Function]}
      title="Show headers"
    >
      Headers
    </button>
    <label
      className="basic-auth"
    >
      <input
        checked={false}
        onChange={[MockFunction]}
        type="checkbox"
      />
       
      Basic Auth
    </label>
  </div>
</div>
`;
